### PR TITLE
INGK-988 Use a dataclass to represent a dictionary error

### PR DESCRIPTION
**Pull request by MartinALL91**
### Description

Use a dataclass to represent a dictionary error.

Fixes # INGK-988

### Type of change

- Create the DictionaryError class.
- Remove the DictionaryErrors class.

### Tests
- Update tests.

### Documentation

- [x] Update docstrings of every function, method or class that change.

### Code formatting and linting

- [x] Use the ruff package to format the code: `ruff format ingenialink tests`.
- [x] Use the ruff package to lint the code: `ruff check ingenialink`.

### Others

- [x] Set fix version field in the Jira issue.

### CODE REVIEW

File: ingenialink/dictionary.py [hunk 1]
```diff
@@ -114,6 +114,24 @@ def labels(self, cat_id: str) -> Dict[str, str]:
         return self._categories[cat_id]
 
 
+@dataclass
+class DictionaryError:
+    id: str
```
Comment (polfeliu) [inline ingenialink/dictionary.py:119]: Why don't you store the parsed int also here?

File: ingenialink/dictionary.py [hunk 2]
```diff
@@ -134,15 +152,16 @@ def load_errors(self) -> None:
...
                 continue
-            self._errors[int(element.attrib["id"], 16)] = [
-                element.attrib["id"],
-                element.attrib["affected_module"],
-                element.attrib["error_type"].capitalize(),
-                label.text,
-            ]
+            error_id = int(element.attrib["id"], 16)
+            error_description = label.text
+            error_type = element.attrib["error_type"].capitalize()
```
Comment (polfeliu) [inline ingenialink/dictionary.py:157]: Parsers should not modify the values of the things they parse.
@GullinBustin Do you know if error type being used anywhere?

File: ingenialink/dictionary.py [hunk 3]
```diff
@@ -213,7 +232,7 @@ class Dictionary(ABC):
     """Number of subnodes in the dictionary."""
     categories: DictionaryCategories
     """Instance of all the categories in the dictionary."""
-    errors: DictionaryErrors
+    errors: Dict[int, "DictionaryError"]
```
Comment (polfeliu) [inline ingenialink/dictionary.py:235]: I'm find with this simplification, but then the DictionaryErrors class serves no purpose, so the parsing method should be moved here

File: ingenialink/dictionary.py (hunk 1 above)
Comment (MartinALL91) [reply ingenialink/dictionary.py:119]: To keep the format as it was before. If we store an int, we will have to make changes in ML3.

File: ingenialink/dictionary.py (hunk 2 above)
Comment (MartinALL91) [reply ingenialink/dictionary.py:157]: The error type is used in the error widget of ML3 for the category field.

File: ingenialink/dictionary.py (hunk 1 above)
Comment (polfeliu) [reply ingenialink/dictionary.py:119]: I would expect the GUI to do the formatting of values so the backwards compatibility only on the __iter__ method, not the new API

**Review by polfeliu — APPROVED** [review]

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**